pkg/controller/worker/machine_dependencies.go

```diff
@@ -59,98 +54,97 @@ func (w *workerDelegate) CleanupMachineDependencies(ctx context.Context) error {
 	if err := shootClient.List(ctx, shootNodes); err != nil {
 		return fmt.Errorf("failed to get shoot nodes: %v", err)
 	}
+
 	// go through each node, for each one without the right annotation, get the private network
-	var targetCidrs []string
+	targetCIDRs := sets.NewString()
 	for _, n := range shootNodes.Items {
-		if n.Annotations == nil || n.Annotations[equinixMetalPrivateNetworkAnnotations] == "" {
+		if n.Annotations[equinixMetalPrivateNetworkAnnotations] == "" {
```

Comment:
Are you sure `n.Annotations` cannot be `nil`?

Reply:
It can be `nil`, but newer Golang versions can cater with it, i.e., no need to handle it explicitly.